File: java/src/main/java/shopping/merchant/samples/accounts/accounts/v1/GetAccountByAliasSample.java
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package shopping.merchant.samples.accounts.accounts.v1beta;

// [START merchantapi_get_account_by_alias]
import com.google.api.gax.core.FixedCredentialsProvider;
import com.google.auth.oauth2.GoogleCredentials;
import com.google.shopping.merchant.accounts.v1beta.Account;
import com.google.shopping.merchant.accounts.v1beta.AccountName;
import com.google.shopping.merchant.accounts.v1beta.AccountsServiceClient;
import com.google.shopping.merchant.accounts.v1beta.AccountsServiceSettings;
import com.google.shopping.merchant.accounts.v1beta.GetAccountRequest;
import shopping.merchant.samples.utils.Authenticator;

/** This class demonstrates how to get a single Merchant Center account by its alias. */
public class GetAccountByAliasSample {

  public static void getAccountByAlias(long providerId, String alias) throws Exception {

    // Obtains OAuth token based on the user's configuration.
    GoogleCredentials credential = new Authenticator().authenticate();

    // Creates service settings using the credentials retrieved above.
    AccountsServiceSettings accountsServiceSettings =
        AccountsServiceSettings.newBuilder()
            .setCredentialsProvider(FixedCredentialsProvider.create(credential))
            .build();

    // Creates account name to identify account.
    // The name has the format: accounts/{providerId}~{alias}
    // This format can used whenever an account name is needed. For example it can also be used to
    // get the homepage of an account or approve, get or list its services etc.
    // For more information about aliases see
    // https://developers.google.com/merchant/api/guides/accounts/relationships
    String name = AccountName.newBuilder().setAccount(providerId + "~" + alias).build().toString();

    // Calls the API and catches and prints any network failures/errors.
    try (AccountsServiceClient accountsServiceClient =
        AccountsServiceClient.create(accountsServiceSettings)) {

      GetAccountRequest request = GetAccountRequest.newBuilder().setName(name).build();

      System.out.println("Sending Get Account request:");
      Account response = accountsServiceClient.getAccount(request);

      System.out.println("Retrieved Account below");
      System.out.println(response);
    } catch (Exception e) {
      System.out.println(e);
    }
  }

  public static void main(String[] args) throws Exception {
    // Update this with the provider ID of the account you want to get.
    long providerId = 123L;
    // Update this with the alias of the account you want to get.
    String alias = "alias";
    getAccountByAlias(providerId, alias);
  }
}
// [END merchantapi_get_account_by_alias]

File: python/examples/accounts/accounts/v1/get_account_by_alias_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""This class demonstrates how to get a single Merchant Center account by its alias."""

# [START merchantapi_get_account_by_alias]
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1 import AccountsServiceClient
from google.shopping.merchant_accounts_v1 import GetAccountRequest


def get_account_by_alias(provider_id: int, alias: str) -> None:
  """Gets a single Merchant Center account by its alias.

  Args:
      provider_id: The provider ID of the account.
      alias: The alias of the account.
  """

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = AccountsServiceClient(credentials=credentials)

  # Creates the name of the account to retrieve.
  # The name has the format: accounts/{providerId}~{alias}
  # This format can be used whenever an account name is needed. For example it
  # can also be used to get the homepage of an account or approve, get or list
  # its services etc.
  # For more information about aliases see
  # https://developers.google.com/merchant/api/guides/accounts/relationships
  name = f"accounts/{provider_id}~{alias}"

  # Creates the request.
  request = GetAccountRequest(name=name)

  # Makes the request and catches and prints any error messages.
  try:
    print("Sending Get Account request:")
    response = client.get_account(request=request)
    print("Retrieved Account below")
    print(response)
  except RuntimeError as e:
    print(e)


if __name__ == "__main__":
  # Update this with the provider ID of the account you want to get.
  provider_id_ = 123
  # Update this with the alias of the account you want to get.
  alias_ = "alias"
  get_account_by_alias(provider_id_, alias_)

# [END merchantapi_get_account_by_alias]
